templates/media-storage.yml

```diff
@@ -0,0 +1,146 @@
+---
+AWSTemplateFormatVersion: 2010-09-09
+Description: >-
+  Create a file storage stack for streaming media
+
+Parameters:
+  MediaBucketName:
+    AllowedPattern: '^[0-9a-zA-Z]+([0-9a-zA-Z-]*[0-9a-zA-Z])*$'
+    ConstraintDescription: >-
+      S3 bucket name can include numbers, lowercase letters, uppercase
+      letters, and hyphens (-). It cannot start or end with a hyphen (-).
+    Description: S3 bucket name used to store media files
+    Default: mpa-media-storage
+    Type: String
+  MetadataTableName:
+    Description: The DynamoDB table name which stores file metadata
+    Type: String
+    Default: Metadata
+
+Resources:
+  MediaBucket:
+    Type: 'AWS::S3::Bucket'
+    DependsOn:
+      - ProcessingFileMetadataPermission
+    Properties:
+      BucketName: !Ref MediaBucketName
+      NotificationConfiguration:
+        LambdaConfigurations:
+          - Event: 's3:ObjectCreated:*'
+            Function: !GetAtt ProcessingFileMetadataLambdaFunction.Arn
+  ProcessingFileMetadataLambdaFunction:
+    Type: 'AWS::Lambda::Function'
+    DependsOn:
+      - ProcessingFileMetadataExecutionRole
+    Properties:
+      Code:
+        ZipFile: !Sub |
+          import boto3
+          import json
+          import logging
+          import os
+          import time
+          import urllib.parse as urllib
+          import uuid
+
+          METADATA_TABLE = os.environ.get("METADATA_TABLE")
```

Comment:
now we're using python to run the lambda function.
it seems we can migrate code to java codebase in the further.